### PR TITLE
Uml 1316 move check lpa handler logic into new class

### PR DESCRIPTION
**Pull request by GemTay**
# Purpose

Refactors the checkLpaHandler by using the new API endpoint which contains the business logic

Fixes UML-1316

## Approach

Created a new class in the front called AddLpa which handles the responses from the new API endpoints /v1/add-lpa/validate and /v1/add-lpa/confirm. It maps the possible responses into new event codes that are returned as part of a custom response object AddLpaApiResponse, which contains the event code and any additional data. The checkLpaHandler uses a switch function to display the correct page based off the event code in the response object. I have added unit tests for the new classes, altered existing test in the front and api accordingly. I have deleted the functions from the LpaService that are no longer used such as getLpaByPasscode. The add an LPA journey remains the same.

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by SeemaMenon — APPROVED** [review]
LGTM

File: service-api/app/features/actor-add-an-lpa.feature [hunk 1]
```diff
@@ -31,9 +30,15 @@ Feature: Add an LPA
     And The LPA has not been added
 
   @integration @acceptance @pact
-  Scenario: The user is shown an error message when attempting to add the same LPA twice
+  Scenario: The user is told when attempting to add the same LPA twice
```
Comment (cooperaj) [inline service-api/app/features/actor-add-an-lpa.feature:33]: Whats this one testing?

File: service-api/app/features/context/Integration/LpaContext.php [hunk 2]
```diff
@@ -149,32 +149,6 @@ public function iAmToldThatIHaveAnActivationKeyForThisLPAAndWhereToFindIt()
      * @When /^I attempt to add the same LPA again$/
      */
     public function iAttemptToAddTheSameLPAAgain()
-    {
-        $this->pactPostInteraction(
```
Comment (cooperaj) [inline service-api/app/features/context/Integration/LpaContext.php:153]: Oh, did we move all the right PACT stuff about?

File: service-front/app/src/Actor/src/Handler/CheckLpaHandler.php [hunk 3]
```diff
@@ -122,58 +128,61 @@ public function handle(ServerRequestInterface $request): ResponseInterface
...
-            if (!is_null($lpaAddedData)) {
+        $result = $this->addLpa->validateAddLpaData(
+            $this->identity,
+            $passcode,
+            $referenceNumber,
+            $dob
+        );
+
+        switch ($result->getResponse()) {
+            case AddLpaApiResponse::ADD_LPA_ALREADY_ADDED:
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Handler/CheckLpaHandler.php:154]: soooooo much better

File: service-front/app/src/Actor/src/Handler/CheckLpaHandler.php [hunk 4]
```diff
@@ -299,4 +263,30 @@ public function handlePost(
...
         ]));
     }
+
+    /**
+     * @param ArrayObject $lpaData
+     *
+     * @return string
+     */
+    private function identifyActorRole(ArrayObject $lpaData): string
+    {
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Handler/CheckLpaHandler.php:273]: Mmm. Is this something we only use here? Seems odd to have it as a part of a handler.

File: service-api/app/features/actor-add-an-lpa.feature (hunk 1 above)
Comment (GemTay) [reply service-api/app/features/actor-add-an-lpa.feature:33]: Its testing that a bad request is thrown with the correct message and additional data if a user tries to add an LPA that they have already added

File: service-api/app/features/context/Integration/LpaContext.php (hunk 2 above)
Comment (GemTay) [reply service-api/app/features/context/Integration/LpaContext.php:153]: I think so, the pact interaction for validating the code comes after the check to see if the user has already added the LPA, so it wasn't needed in this step anymore

File: service-front/app/src/Actor/src/Handler/CheckLpaHandler.php (hunk 4 above)
Comment (GemTay) [reply service-front/app/src/Actor/src/Handler/CheckLpaHandler.php:273]: Yeah it's only used here, i'll see if I could move it into the AddLpa class

File: service-front/app/src/Actor/src/Handler/CheckLpaHandler.php [hunk 5]
```diff
@@ -180,9 +181,18 @@ public function handleGet(
                 );
             case AddLpaApiResponse::ADD_LPA_FOUND:
                 $lpaData = $result->getData();
-                $actorRole = $this->identifyActorRole($lpaData);
                 $lpa = $lpaData['lpa'];
                 $actor = $lpaData['actor']['details'];
+                $actorRole = ($lpaData['actor']['type'] === 'donor') ? 'Donor' : 'Attorney';
```
Comment (GemTay) [inline service-front/app/src/Actor/src/Handler/CheckLpaHandler.php:186]: this comes from the ResolveActor class in the api

File: service-front/app/src/Actor/src/Handler/CheckLpaHandler.php [hunk 6]
```diff
@@ -255,40 +235,34 @@ public function handlePost(
         $this->form->setData($request->getParsedBody());
 
         if ($this->form->isValid()) {
-            $actorCode = $this->lpaService->confirmLpaAddition(
+            $result = $this->addLpa->confirmLpaAddition(
```
Comment (cooperaj) [inline service-front/app/src/Actor/src/Handler/CheckLpaHandler.php:238]:
```suggestion
            $result = $this->addLpa->confirm(
```
Just a thought.

Comment (GemTay) [reply service-front/app/src/Actor/src/Handler/CheckLpaHandler.php]: good point, i prefer 'confirm', and i've changed the validate func to just 'validate' too for consistency

**Review by cooperaj — APPROVED** [review]

**Pull request merged**